components/edit_post_modal/edit_post_modal.test.tsx

```diff
@@ -0,0 +1,863 @@
+            openModal: jest.fn(),
+            setShowPreview: jest.fn(),
+        },
+    }: Props,
+) {
+    return (
+        <EditPostModal
+            canEditPost={canEditPost}
+            shouldShowPreview={false}
+            canDeletePost={canDeletePost}
+            ctrlSend={ctrlSend || false}
+            config={config}
+            editingPost={editingPost}
+            actions={actions}
+            maxPostSize={Constants.DEFAULT_CHARACTER_LIMIT}
+            useChannelMentions={useChannelMentions}
+        />
+    );
+}
+
+describe('components/EditPostModal', () => {
+    it('should match with default config', () => {
+        const wrapper = shallowWithIntl(createEditPost({...defaultProps}));
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should match without emoji picker', () => {
+        const config = {
+            PostEditTimeLimit: 300,
+            EnableEmojiPicker: 'false',
+        };
+        const wrapper = shallowWithIntl(createEditPost({...defaultProps, config}));
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should not call openModal on empty edited message but with attachment', () => {
+        window.scrollTo = () => {};
+        const actions = {
+            editPost: jest.fn((data) => (Promise.resolve(data))),
+            addMessageIntoHistory: jest.fn(),
+            hideEditPostModal: jest.fn(),
+            openModal: jest.fn(),
+            setShowPreview: jest.fn(),
+        };
+        const editingPost = {
+            postId: '123',
+            post: {
+                ...defaultPost,
+                file_ids: ['file_id_1'],
+            },
+            commentCount: 3,
+            refocusId: 'test',
+            show: true,
+            title: 'test',
+        };
+
+        const wrapper = shallowWithIntl(createEditPost({...defaultProps, actions, editingPost}));
+        const instance = wrapper.instance() as EditPostModalClass;
+        wrapper.setState({editText: ''});
+        instance.handleEdit();
+
+        expect(actions.openModal).not.toHaveBeenCalled();
+        expect(actions.addMessageIntoHistory).toBeCalled();
+        expect(actions.editPost).toBeCalled();
+    });
+
+    it('should call editPost, addMessageIntoHistory and hideEditPostModal on save', async () => {
+        window.scrollTo = () => {};
+        const actions = {
+            editPost: jest.fn((data) => (Promise.resolve(data))),
+            addMessageIntoHistory: jest.fn(),
+            hideEditPostModal: jest.fn(),
+            openModal: jest.fn(),
+            setShowPreview: jest.fn(),
+        };
+        const wrapper = shallowWithIntl(createEditPost({...defaultProps, actions}));
+
+        expect(actions.addMessageIntoHistory).not.toBeCalled();
+        expect(actions.editPost).not.toBeCalled();
+
+        wrapper.setState({editText: 'new message'});
+        wrapper.find('.btn-primary').simulate('click');
+
+        await runAllPromises();
```

Comment:
How come we are replacing the original await with this?

Reply:
Thanks, I missed this from the original

Reply:
The tests were failing in CI but passing locally without this change. I reverted it for now because it feels hacky but as you can see the CI tests are failing again. Do you have an idea what is wrong?

Reply:
Honestly not sure myself. I agree though that it feels hacky. I'll ask in MM to see if anyone has any insight.

Reply:
Did you discover anything about this issue @devinbinnie ?

Reply:
Unfortunately not, so let's go with the hack for now and put a comment on it. Sorry for the slow response.